### ISSUE TITLE
[java]ReturnEmptyArrayRatherThanNull can not detect return null surrounded by if statement

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#2540]
**Affects PMD Version:** 
6.22.0

**Rule:**
ReturnEmptyArrayRatherThanNull

**Description:**
ReturnEmptyArrayRatherThanNull can not detect <code>return null</code> surrounded by if statement
AST analyse result of the code sample:
```
//MethodDeclaration
[
(./ResultType/Type[@ArrayType='true'])
and
(./Block/BlockStatement/Statement/ReturnStatement/Expression/PrimaryExpression/PrimaryPrefix/Literal/NullLiteral)
]
```
AST chain to <code>return null</code> in the following code sample is: Block -> BlockStatement -> Statement -> IfStatement -> Statement  -> ReturnStatement  -> ...  -> NullLiteral

**Code Sample demonstrating the issue:**
```
public byte []readBytes() throws IOException {
	int tag = is.read();
    if (tag == 'N')
		return null;
```

**Expected outcome:**
false-negative

**Running PMD through:**
CLI

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
